src/recipients/RecipientItemV2.tsx

```diff
@@ -14,19 +21,26 @@ interface Props {
   recipient: Recipient
   onSelectRecipient(recipient: Recipient): void
   loading: boolean
-  showValoraIcon?: boolean
   selected?: boolean
 }
 
 const ICON_SIZE = 10
 
-function RecipientItem({ recipient, onSelectRecipient, loading, showValoraIcon, selected }: Props) {
+function RecipientItem({ recipient, onSelectRecipient, loading, selected }: Props) {
   const { t } = useTranslation()
 
   const onPress = () => {
     onSelectRecipient(recipient)
   }
 
+  const e164NumberToAddress = useSelector(e164NumberToAddressSelector)
```

Comment:
One downside of using this state is that whenever we lookup a PN mapping for a contact that has already been looked up and has an address mapping, the mapping is cleared before being populated again once we get back a response. This causes a small flash of the icon (see 3rd contact click on the screen recording attached). One option to avoid this is to have a separate list of known valora contacts in redux state which is populated along with the `e164NumberToAddress` state. We can't change the behavior of `e164NumberToAddress` state because the value being `undefined` acts as a proxy for "loading" state for a PN being looked up. Thoughts?

Reply:
yeah we could track that differently but that would mean changes to the existing recipient screen, since we use this state to show the loading indicator next to a recipient. I'll write up an issue for this with the couple of ideas we have.